src/darwin/Framework/CHIP/CHIPDeviceController.mm

```diff
@@ -518,15 +517,14 @@ - (NSString *)openPairingWindowWithPIN:(uint64_t)deviceID
             *error = [CHIPError errorForCHIPErrorCode:CHIP_ERROR_INVALID_INTEGER_VALUE];
         }
         return nil;
-    } else {
-        setupPayload.discriminator = (uint16_t) discriminator;
     }
 
     setupPIN &= ((1 << chip::kSetupPINCodeFieldLengthInBits) - 1);
-    setupPayload.setUpPINCode = (uint32_t) setupPIN;
 
-    err = self.cppCommissioner->OpenCommissioningWindow(deviceID, (uint16_t) duration, 1000, (uint16_t) discriminator,
-        chip::Controller::DeviceController::CommissioningWindowOption::kTokenWithProvidedPIN, setupPayload);
+    chip::SetupPayload setupPayload;
+    err = chip::Controller::AutoCommissioningWindowOpener::OpenCommissioningWindow(self.cppCommissioner, deviceID,
+        chip::System::Clock::Seconds16(static_cast<uint16_t>(duration)), 1000, static_cast<uint16_t>(discriminator),
```

Comment:
What is the magic 1000 ?

Reply:
Please move to use `Crypto::kSpake2p_Min_PBKDF_Iterations`

Reply:
Added a comment.